src/error.rs

```diff
@@ -9,6 +9,10 @@ pub enum ExternalError {
     NotSupported(NotSupportedError),
     /// The OS cannot perform the operation.
     Os(OsError),
+    /// The event loop can't be re-run while it's already running
+    AlreadyRunning,
+    /// Application has exit with an error status.
+    ExitFailure(i32),
```

Comment:
Not entirely sure what's best here.

Filling out `EventLoopClosed` then adds some seemingly unrelated errors for `EventLoopProx::send_event()`

`AlreadyRunning` feels like it fits with "An error whose cause it outside Winit's control" - since the developer is trying to do something that's documented as unsupported and we have to bail due to actions by the developer that we can't control.

I'm less sure about `ExitFailure(i32)` but it kind of is an error that's outside of Winit's control too - it's an error that's being forwarded from the app, so maybe that's still ok.

Looking at where `ExternalError` is used though makes me wonder if that could be renamed to `RequestError` to convey that they are errors you might expect while making requests to query state. (though wouldn't do that as part of this PR)

Maybe we could add a very general `RuntimeError` for all the loop `run` APIs to use which would look like this current `ExternalError`? (and leave `ExternalError` + `EventLoopCleared` as they are for now)

Reply:
>Maybe we could add a very general RuntimeError for all the loop run APIs to use which would look like this current ExternalError? (and leave ExternalError + EventLoopCleared as they are for now)

Well, yeah, having special error for event loop is what I want, the `ExternalError` sounds weird indeed, though. It should be named RequestError, because that's when it's being raised.

Reply:
I'm currently adding a `RunLoopError` to use

Reply:
Maybe something like `EventLoopError` and we could later you it for creation as well or do we really want to have a separate error type just for the run APIs?

Reply:
It was a pretty big faff to change by rebasing the change through all the separate patches.

If we want to change the name again then I can just do this as a follow up commit to rename.

Reply:
You can squash all your changes into one commit with a nice description, since I'll likely squash everything anyway. I don't see a value in having separate commits like that, since it'll likely won't build in-between or work in some cases in between. So better to apply everything at once in a one commit.

Reply:
As discussed on IRC - I'm hoping you don't end up squashing these commits since I intentionally created a series of patches that made the changes in logical steps (in part so we could have the option to split the PR up).

The patches each have commit messages I wouldn't want to throw away and each patch should be able to pass CI.